--- ./src/slideshow.c.org	2011-09-01 11:24:04.424019451 +0200
+++ ./src/slideshow.c	2011-09-01 11:23:57.240352664 +0200
@@ -171,15 +171,7 @@ void feh_reload_image(winwidget w, int r
 
 	/* if the image has changed in dimensions - we gotta resize */
 	if ((feh_load_image(&tmp, FEH_FILE(w->file->data))) == 0) {
-		if (force_new) {
-			eprintf("failed to reload image\n");
-		} else {
-			im_weprintf(w, "Couldn't reload image. Is it still there?");
-		}
-		winwidget_rename(w, title);
-		free(title);
-		free(new_title);
-		filelist = feh_file_remove_from_list(filelist, w->file);
+		weprintf("failed to reload image\n");
 		return;
 	}
 
